// laravel-nova-ui/tailwind.config.cjs
const defaultTheme = require('tailwindcss/defaultTheme')
const { generateRootCSSVars, generateTailwindColors } = require('./generators')
import tailwindConfig from '@hempworks/pilgrim/tailwind.config'

module.exports = {
  mode: 'jit',
  presets: [tailwindConfig],
  content: [
    './components/**/*.vue',
    './src/**/*.{tsx,vue}',
    './docs/**/*.md',
    './.vitepress/**/*.vue',
    ...tailwindConfig.content,
  ],

  safelist: [
    // {
    //   pattern: /bg-(?:\w)+-(?:\d)+/,
    //   variants: ['sm', 'md', 'lg'],
    // },
    // {
    //   pattern: /text-(?:\w)+-(?:\d)+/,
    //   variants: ['sm', 'md', 'lg'],
    // },
    // {
    //   pattern: /grid-cols-(?:\d)+/,
    //   variants: ['sm', 'md', 'lg'],
    // },
    // {
    //   pattern: /gap-(?:\d)+/,
    //   variants: ['sm', 'md', 'lg'],
    // },
    // {
    //   pattern: /col-span-(?:\d)+/,
    //   variants: ['sm', 'md', 'lg'],
    // },
  ],

  theme: {
    colors: generateTailwindColors(),
    extend: {
      fontFamily: { sans: ['Nunito Sans', ...defaultTheme.fontFamily.sans] },
      fontSize: { xxs: '11px' },
      maxWidth: { xxs: '15rem' },
      minHeight: theme => theme('spacing'),
      minWidth: theme => theme('spacing'),

      spacing: {
        5: '1.25rem',
        9: '2.25rem',
        11: '2.75rem',
      },

      top: theme => theme('inset'),
      width: theme => theme('spacing'),
    },
  },
  plugins: [
    require('@tailwindcss/typography'),
    function ({ addBase }) {
      addBase({ ':root': generateRootCSSVars() })
    },
  ],
}
