// laravel-nova-ui/dist/Components/Table/Table.vue.cjs
'use strict';

Object.defineProperties(exports, { __esModule: { value: true }, [Symbol.toStringTag]: { value: 'Module' } });

const vue = require('vue');
const Loader_vue_vue_type_script_setup_true_lang = require('../Loader/Loader.vue.cjs');

const _hoisted_1 = { class: "relative overflow-hidden overflow-x-auto" };
const _hoisted_2 = { class: "absolute inset-0 z-10 bg-white opacity-75 dark:bg-gray-950" };
const _hoisted_3 = { key: 0 };
const _hoisted_4 = { class: "py-6" };
const _hoisted_5 = { key: 1 };
const _hoisted_6 = {
  key: 2,
  class: "divide-y divide-gray-100 dark:divide-gray-700"
};
const _hoisted_7 = {
  key: 0,
  class: "w-full divide-y divide-gray-100 dark:divide-gray-700"
};
const _sfc_main = /* @__PURE__ */ vue.defineComponent({
  __name: "Table",
  props: {
    loading: { type: Boolean, default: false },
    refreshing: { type: Boolean, default: false },
    empty: { type: Boolean, default: false }
  },
  setup(__props) {
    return (_ctx, _cache) => {
      return vue.openBlock(), vue.createElementBlock("div", null, [
        vue.createElementVNode("div", _hoisted_1, [
          vue.withDirectives(vue.createElementVNode("div", _hoisted_2, null, 512), [
            [vue.vShow, _ctx.refreshing]
          ]),
          _ctx.loading ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_3, [
            vue.createElementVNode("div", _hoisted_4, [
              vue.createVNode(Loader_vue_vue_type_script_setup_true_lang.default, { class: "text-gray-400" })
            ])
          ])) : vue.createCommentVNode("", true),
          _ctx.empty ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_5, [
            vue.renderSlot(_ctx.$slots, "empty")
          ])) : vue.createCommentVNode("", true),
          !_ctx.loading && !_ctx.empty ? (vue.openBlock(), vue.createElementBlock("div", _hoisted_6, [
            vue.renderSlot(_ctx.$slots, "toolbar"),
            !_ctx.loading && !_ctx.empty ? (vue.openBlock(), vue.createElementBlock("table", _hoisted_7, [
              vue.renderSlot(_ctx.$slots, "default")
            ])) : vue.createCommentVNode("", true),
            vue.renderSlot(_ctx.$slots, "footer")
          ])) : vue.createCommentVNode("", true)
        ])
      ]);
    };
  }
});

exports.default = _sfc_main;
