// lodash/keysIn.cjs
'use strict';

Object.defineProperty(exports, Symbol.toStringTag, { value: 'Module' });

const _arrayLikeKeys = require('./_arrayLikeKeys.cjs');
const _baseKeysIn = require('./_baseKeysIn.cjs');
const isArrayLike$1 = require('./isArrayLike.cjs');

var arrayLikeKeys = _arrayLikeKeys._arrayLikeKeys,
    baseKeysIn = _baseKeysIn._baseKeysIn,
    isArrayLike = isArrayLike$1.isArrayLike_1;

/**
 * Creates an array of the own and inherited enumerable property names of `object`.
 *
 * **Note:** Non-object values are coerced to objects.
 *
 * @static
 * @memberOf _
 * @since 3.0.0
 * @category Object
 * @param {Object} object The object to query.
 * @returns {Array} Returns the array of property names.
 * @example
 *
 * function Foo() {
 *   this.a = 1;
 *   this.b = 2;
 * }
 *
 * Foo.prototype.c = 3;
 *
 * _.keysIn(new Foo);
 * // => ['a', 'b', 'c'] (iteration order is not guaranteed)
 */
function keysIn(object) {
  return isArrayLike(object) ? arrayLikeKeys(object, true) : baseKeysIn(object);
}

var keysIn_1 = keysIn;

exports.keysIn_1 = keysIn_1;
